### PR TITLE
Cn write s3 delete for single table

### PR DESCRIPTION
**Pull request by JackTan25**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #7772 #7882

## What this PR does / why we need it:
This pr is used to support distributed delete for cn-write-s3.
we just support delete sql for single table.
1. dispatch all data to every cn node.
2. in each node, it will just delete data belong to itself (we use group operator to implement bucket operations, we need to make sure one block will be deleted by one and same cn ).
3. for main cn (which generates the txn), it will hold the deletes info for cn block and dn block, and main cn will process
cn block deletes, dn block deletes, raw batch data, but for remote cn, it will just flush dn block deletes for now, we will improve this in the next pr.

### CODE REVIEW

**Comment by JackTan25** [comment]
I need to test more, the distributed delete is not be tested. I will add bvt test later

File: pkg/container/batch/batch.go [hunk 1]
```diff
@@ -147,6 +147,32 @@ func (bat *Batch) ExpandNulls() {
 	}
 }
 
+func (bat *Batch) AntiShrink(sels []int64) {
```
Comment (fengttt) [inline pkg/container/batch/batch.go:150]: What is an AntiShrink?

**Review by fengttt — CHANGES_REQUESTED** [review]
You don't have a meaningful commit message on such a big PR.   Cannot let this go in.

**Review by aressu1985 — APPROVED** [review]

File: pkg/container/batch/batch.go (hunk 1 above)
Comment (JackTan25) [reply pkg/container/batch/batch.go:150]: A Shrink is used to perserve the rows with sels row number, and the AntiShrink us used to perserve the rows whose row number is not in sels.

**Comment by JackTan25** [comment]
I'm doing test, don't merge it

**Review by reusee — APPROVED** [review]

**Review by heni02 — APPROVED** [review]

**Review by aressu1985 — APPROVED** [review]

**Review by fengttt — APPROVED** [review]

**Review by reusee — APPROVED** [review]

**Review by nnsgmsone — APPROVED** [review]

**Review by triump2020 — APPROVED** [review]

**Review by XuPeng-SH — APPROVED** [review]

**Review by m-schen — APPROVED** [review]

**Pull request merged**